### PR TITLE
chore: react hook form example

### PR DESCRIPTION
**Pull request by Mr0Bread**

### CODE REVIEW

File: examples/with-react-hook-form/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Minimal Next.js App Router example for UploadThing
```
Comment (markflorkowski) [inline examples/with-react-hook-form/README.md:1]: Title should probably mention react-hook-form

File: examples/with-react-hook-form/src/app/page.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  const onSubmit: SubmitHandler<Inputs> = async (data) => {
+    if (!files) return;
+
+    const res = await startUpload(files);
+
+    if (!res) {
+      alert("Upload failed");
+      return;
+    }
+
```
Comment (markflorkowski) [inline examples/with-react-hook-form/src/app/page.tsx:33]: I'd maybe leave some comments in here explaining what information we have at different points (eg. file itself, uploaded url, other form data) and maybe a mock of how a user might want to store info in their db.

**Review by markflorkowski — COMMENTED** [review]
Looks good overall, just left a few comments. 

We may also want an example where files upload immediately, and urls are included in the form data. (Would be a bit more complex since it should probably handle deleting files if the user decides to replace their selection)

**Comment by Mr0Bread** [comment]
@markflorkowski would you like that more advanced use case within this example or in separate one?

**Comment by markflorkowski** [comment]
Hmm, I think it can be in this one, since they are both rhf. Do you think that would be confusing?

**Comment by Mr0Bread** [comment]
@markflorkowski Hm, I've just been thinking. Maybe it's better to simply improve current example? I mean does it make sense to upload the image each time it is dropped? What might be the reason for that? Browser preview does not require it. If form data needs an URL of the image, it can be received during form submission. Simply upload image inside of `onSubmit` and get URL from the response. Let me know, if I'm missing something here

**Comment by markflorkowski** [comment]
The only reason I was thinking was it is slightly better UX in the form to have files upload in the background as you continue to fill out the form, instead of waiting for files to upload when submitting.

In the contrived example of a form that *only* contains a file upload it makes less sense, but I feel like in most cases it will be one field of many on a form.

Regardless, the url should be submitted with the form data, as presumably that data will be stored or used in some way by the application

File: examples/with-react-hook-form/tailwind.config.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { fontFamily } from "tailwindcss/defaultTheme";
+
+import { withUt } from "uploadthing/tw";
+
+/**
+ * @see https://docs.uploadthing.com/theming#with-tailwindcss
```
Comment (juliusmarminge) [inline examples/with-react-hook-form/tailwind.config.ts:6]: https://docs.uploadthing.com/getting-started/appdir#add-uploadthings-styles

File: examples/with-react-hook-form/src/app/page.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  const { startUpload } = useUploadThing("videoAndImage");
+  const { handleSubmit, register, reset } = useForm<Inputs>();
+  const { getInputProps, getRootProps } = useDropzone({
+    onDrop: (files) => {
+      setFiles(files);
+    },
+  });
+
+  const onSubmit: SubmitHandler<Inputs> = async (data) => {
+    if (!files) return;
```
Comment (juliusmarminge) [inline examples/with-react-hook-form/src/app/page.tsx:25]: why aren't files set on the form?

File: examples/with-react-hook-form/README.md (hunk 1 above)
Comment (chihabhajji) [reply examples/with-react-hook-form/README.md:1]: i crawled all the repo looking for a react-hook-form implementation, thank you for the suggestion

**Comment by t3dotgg** [comment]
Due to the likelihood of this code being outdated, I'm gonna close and rewrite this example in near future. Sorry for delays on it 🙃

**Comment by DevTGhosh** [comment]
@t3dotgg  looking forward to it.

**Pull request closed without merging**